lib/features/composer/presentation/widgets/saving_message_dialog_view.dart

```diff
@@ -0,0 +1,243 @@
+
+import 'package:core/presentation/extensions/capitalize_extension.dart';
+import 'package:core/presentation/extensions/color_extension.dart';
+import 'package:core/presentation/state/failure.dart';
+import 'package:core/presentation/state/success.dart';
+import 'package:core/presentation/views/button/tmail_button_widget.dart';
+import 'package:core/utils/app_logger.dart';
+import 'package:dartz/dartz.dart' as dartz;
+import 'package:dio/dio.dart';
+import 'package:flutter/material.dart';
+import 'package:get/get.dart';
+import 'package:tmail_ui_user/features/composer/domain/exceptions/compose_email_exception.dart';
+import 'package:tmail_ui_user/features/composer/domain/state/generate_email_state.dart';
+import 'package:tmail_ui_user/features/composer/domain/state/save_email_as_drafts_state.dart';
+import 'package:tmail_ui_user/features/composer/domain/state/update_email_drafts_state.dart';
+import 'package:tmail_ui_user/features/composer/domain/usecases/create_new_and_save_email_to_drafts_interactor.dart';
+import 'package:tmail_ui_user/features/composer/presentation/model/create_email_request.dart';
+import 'package:tmail_ui_user/main/exceptions/remote_exception.dart';
+import 'package:tmail_ui_user/main/localizations/app_localizations.dart';
+import 'package:tmail_ui_user/main/routes/route_navigation.dart';
+
+typedef OnCancelSavingEmailToDraftsAction = Function({CancelToken? cancelToken});
+
+class SavingMessageDialogView extends StatefulWidget {
+
+  final CreateEmailRequest createEmailRequest;
+  final CreateNewAndSaveEmailToDraftsInteractor createNewAndSaveEmailToDraftsInteractor;
+  final OnCancelSavingEmailToDraftsAction? onCancelSavingEmailToDraftsAction;
+  final CancelToken? cancelToken;
+
+  const SavingMessageDialogView({
+    super.key,
+    required this.createEmailRequest,
+    required this.createNewAndSaveEmailToDraftsInteractor,
+    this.onCancelSavingEmailToDraftsAction,
+    this.cancelToken,
+  });
+
+  @override
+  State<SavingMessageDialogView> createState() => _SavingMessageDialogViewState();
+}
+
+class _SavingMessageDialogViewState extends State<SavingMessageDialogView> {
+
+  StreamSubscription? _streamSubscription;
+  final ValueNotifier<dartz.Either<Failure, Success>?> _viewStateNotifier = ValueNotifier(null);
+
+  @override
+  void initState() {
+    super.initState();
+    _streamSubscription = widget.createNewAndSaveEmailToDraftsInteractor
+      .execute(
+        createEmailRequest: widget.createEmailRequest,
+        cancelToken: widget.cancelToken
+      )
+      .listen(
+        _handleDataStream,
+        onError: _handleErrorStream
+      );
+  }
+
+  void _handleDataStream(dartz.Either<Failure, Success> newState) {
+    _viewStateNotifier.value = newState;
+
+    newState.fold(
+      (failure) {
+        if (failure is SaveEmailAsDraftsFailure ||
+            failure is UpdateEmailDraftsFailure ||
+            failure is GenerateEmailFailure) {
+          popBack(result: failure);
+        }
```

Comment:
how about other `failure`?

Reply:
We do not have other failures in this case.